### PR TITLE
Add search! (and other minor upgrades)

### PR DESCRIPTION
**Pull request by mickmcgrath13**
* bump bit-docs-generate-html dependency version
  * https://github.com/bit-docs/bit-docs-generate-html/releases/tag/v0.5.0
  * https://github.com/bit-docs/bit-docs-generate-html/releases/tag/v0.6.0
* bump bit-docs-html-canjs dependency version
  * https://github.com/canjs/bit-docs-html-canjs/releases/tag/v0.5.1
  * https://github.com/canjs/bit-docs-html-canjs/releases/tag/v0.5.2
  * https://github.com/canjs/bit-docs-html-canjs/releases/tag/v0.5.3
  * https://github.com/canjs/bit-docs-html-canjs/releases/tag/v0.5.4
  * https://github.com/canjs/bit-docs-html-canjs/releases/tag/v0.6.0
  * https://github.com/canjs/bit-docs-html-canjs/releases/tag/v0.7.0
  * https://github.com/canjs/bit-docs-html-canjs/releases/tag/v0.7.1
* add bit-docs-generate-searchmap dependency
  * https://github.com/bit-docs/bit-docs-generate-searchmap/releases/tag/v0.0.1-0
* bump bit-docs-generate-searchmap dependency
  * https://github.com/bit-docs/bit-docs-generate-searchmap/releases/tag/v0.1.0
  * https://github.com/bit-docs/bit-docs-generate-searchmap/releases/tag/v0.1.1
* add steal-stache as a bit-docs html dependency

### CODE REVIEW

**Comment by mickmcgrath13** [comment]
Some issues found so far:
* [ ] The `href` attributes in the search results aren't being generated properly
* [ ] When going from one page to another (pushState?), the search disappears

**Comment by mickmcgrath13** [comment]
The href disappearance was due to improper scoping during the es6 -> es5 conversion (changed something to `self` that should have stayed `this`).

The search was disappearing because the `init` is run every time we "change pages" (pushState) - which re-initializes the search.  When the search is initialized, it makes an ajax request to `window.pathPrefix + '/searchMapHash.json'`.

..the problem is that the `window.pathPrefix` exists within a script tag on the newly ajax-retrieved page, but the value isn't making it onto the window (not being executed?) prior to the search being initialized.

While we should fix that (somehow?), it did highlight an opportunity which is: if we do not find the searchMapHash but we do have one in localStorage, we can just use that and continue as normal.

**Comment by mickmcgrath13** [comment]
Maybe a fix could be instead of doing this:
```
<script type="text/javascript">
  window.pathPrefix = '{{pathToDest}}';
</script>
```

...we use a data attribute on a div:
```
<div data-path-prefix="{{pathToDest}}" />
```
..or on the body
```
<body data-path-prefix="{{pathToDest}}">
...
</body>
```

**Comment by matthewp** [comment]
That solution sounds reasonable to me. I would check and see if the body gets updated when going from one page to another... if not putting it on a div might be a better way to go.

**Comment by mickmcgrath13** [comment]
[Previous two issues](https://github.com/canjs/canjs/pull/3262#issuecomment-302528798) have solutions

Now, there's:
[ ] Responsive search

**Comment by mickmcgrath13** [comment]
Issues mentioned above:
* [ ] [broken links](https://github.com/canjs/bit-docs-html-canjs/issues/249)
* [ ] [responsive search](https://github.com/canjs/bit-docs-html-canjs/issues/250)

**Comment by mickmcgrath13** [comment]
https://github.com/canjs/bit-docs-html-canjs/pull/251

**Comment by mickmcgrath13** [comment]
..so, now there's another issue... :/

With the latest update (#250 mentioned in the last comment), the `doc/` isn't at the beginning of the search results as it should be (seems to work in `bit-docs-html-canjs`...)

File: package.json [hunk 1]
```diff
@@ -167,6 +175,6 @@
       "2.3.31": "https://v2.canjs.com"
     },
     "parent": "canjs",
-    "minifyBuild": true
+    "minifyBuild": false
```
Comment (chasenlehara) [inline package.json]: This will need to be flipped back to `true`

File: package.json [hunk 2]
```diff
@@ -144,13 +144,21 @@
...
+      "templates": []
+    },
     "dependencies": {
       "bit-docs-glob-finder": "^0.0.5",
       "bit-docs-dev": "^0.0.3",
       "bit-docs-js": "^0.0.6",
       "bit-docs-tag-sourceref": "^0.0.3",
-      "bit-docs-generate-html": "^0.4.1",
-      "bit-docs-html-canjs": "^0.4.15",
+      "bit-docs-generate-html": "^0.5.0",
```
Comment (chasenlehara) [inline package.json]: `^0.6.0` now

**Comment by justinbmeyer** [comment]
it seems like this checks in the `docMap.json`

**Comment by justinbmeyer** [comment]
- the sidebar isn't dynamic width (it also is janky as the page is loading).
- the search doesn't appear to be in the right spot

cursor_and_canjs

- this can't be published without fixing the ability to show nested children:

cursor_and_canjs_-_api_docs

Are these structural changes in master?  If not, they should be reverted.  We shouldn't publish these structural changes accidentally.

**Comment by chasenlehara** [comment]
The commits from this branch were merged but the search feature isn’t yet active; I will open a new PR to enable search sometime in the future.

**Pull request merged**